generated/dist-esm/models/index.d.ts

```diff
@@ -8,14 +8,38 @@ export interface Conversation {
     metadata?: Record<string, unknown>;
 }
 export interface Message {
-    role: MessageRole;
-    message: string;
+    type?: MessageType;
     eventTimestamp?: Date;
+    role?: MessageRole;
+    message?: string;
+    /** Any object */
+    metadata?: Record<string, unknown>;
     rating?: Rating;
+    name?: string;
+    thought?: string;
+    /** Any object */
+    input?: Record<string, unknown>;
+    observation?: string;
 }
```

Comment:
I am not sure if that's right. Shouldn't we have two new interfaces that extends base interface? This interface makes it completely legal to have both role and name defined on a single object

Reply:
Yup.... turns out autorest does not allow OneOf . This means that you can't mix and match in the messages array, you can only have one type. Not ideal but for the moment all messages will be of one type ...